lib/Models/SearchProviders/NominatimSearchProvider.ts

```diff
@@ -0,0 +1,140 @@
+    super(uniqueId, terria);
+
+    makeObservable(this);
+  }
+
+  protected logEvent(searchText: string) {
+    this.terria.analytics?.logEvent(
+      Category.search,
+      SearchAction.nominatim,
+      searchText
+    );
+  }
+
+  protected doSearch(
+    searchText: string,
+    searchResults: SearchProviderResults
+  ): Promise<void> {
+    searchResults.results.length = 0;
+    searchResults.message = undefined;
+
+    if (searchText === undefined || /^\s*$/.test(searchText)) {
+      return Promise.resolve();
+    }
+
+    const view = this.terria.currentViewer.getCurrentCameraView();
+    const bboxStr =
+      CesiumMath.toDegrees(view.rectangle.west) +
+      ", " +
+      CesiumMath.toDegrees(view.rectangle.north) +
+      ", " +
+      CesiumMath.toDegrees(view.rectangle.east) +
+      ", " +
+      CesiumMath.toDegrees(view.rectangle.south);
+
+    const promise = loadJson(
+      this.url +
+        "search?q=" +
+        searchText +
+        "&viewbox=" +
+        bboxStr +
+        "&bounded=0" +
+        "&format=geojson" +
+        "&countrycodes=" +
+        this.countryCodes +
+        "&limit=" +
+        this.maxResults
+    );
+    return promise
+      .then((result) => {
+        if (searchResults.isCanceled) {
+          // A new search has superseded this one, so ignore the result.
+          return;
+        }
+
+        if (
+          !result?.features ||
+          !Array.isArray(result.features) ||
+          result.features.length === 0
+        ) {
+          searchResults.message = {
+            content: "translate#viewModels.searchNoLocations"
+          };
+          return;
+        }
+
+        const locations: SearchResult[] = (result.features as Feature<Point>[])
+          .filter(
+            (feat) =>
+              feat.properties && feat.geometry && feat.properties.display_name
+          )
+          .sort((a, b) => b.properties!.importance - a.properties!.importance)
+          .map((feat) => {
+            return new SearchResult({
+              name: feat.properties!.display_name,
+              clickAction: createZoomToFunction(this, feat),
+              location: {
+                latitude: feat.geometry.coordinates[1],
+                longitude: feat.geometry.coordinates[0]
+              }
+            });
+          });
+
+        runInAction(() => {
+          searchResults.results.push(...locations);
+        });
+
+        if (searchResults.results.length === 0) {
+          searchResults.message = {
+            content: "translate#viewModels.searchNoLocations"
+          };
+        }
+      })
+      .catch(() => {
+        if (searchResults.isCanceled) {
+          // A new search has superseded this one, so ignore the result.
+          return;
+        }
+
+        searchResults.message = {
+          content: "translate#viewModels.searchErrorOccurred"
+        };
+      });
+  }
+}
+
+function createZoomToFunction(model: NominatimSearchProvider, resource: any) {
```

Comment:
Can you also add the proper type definition to the resource here instead of any? It should be Feature<Point> if I am correct.